### ISSUE TITLE
autocomplete on mobile/android: virtual keyboard not appearing

### ISSUE THREAD

**Issue by matthesrieke** [angular/material#11339]
#### [CodePen Demo](http://codepen.io/team/AngularMaterial/pen/bEGJdd) which demonstrates the issue: 

https://codepen.io/anon/pen/mKjXMN (default CodePen created from the "Floating Label" example in https://material.angularjs.org/1.1.9/demo/autocomplete)

#### Detailed Reproduction Steps: 
1. Open up https://material.angularjs.org/1.1.9/demo/autocomplete on Chrome in Android (tested with two Android 7.0 stock ROMs (Huawei Honor 9 and HTC One M9)
1. Scroll to the floating label example and tap the "Favorite State" input
1. Type something that matches (e.g. "Ari")
1. Tap outside of the input and suggestions
1. Wait 2-3 seconds
1. Tap again into the "Favorite State" input

### What is the expected behavior?

The Virtual Keyboard should open up when tapping into the input field

### What is the current behavior?

The Virtual Keyboard does not appear _most of the time_. When tapping outside and again inside within a short period (< 1 second) the virtual keyboard more reliably appears.

### What is the use-case or motivation for changing an existing behavior?

Bug.

### Which versions of AngularJS, Material, OS, and browsers are affected?

- AngularJS: 1.6.7
- AngularJS Material: 1.1.9
- OS: Android 7.0 stock ROMs (Huawei Honor 9 and HTC One M9)
- Browsers: Chrome (Brave also applies)

**Comment by Splaktar** [comment]
I am not able to reproduce this using a Pixel XL running Android 8.1.0 with Chrome or Firefox. I tried quick pauses and long pauses, but the virtual keyboard appeared almost instantly each time.

Please provide more details on the reproduction and some indication of how this is a bug in AngularJS Material.

**Comment by matthesrieke** [comment]
Thanks for testing the issue. I can only give you the information I had in my original posting. The one difference I see is the Android version. I can only report it for 7.0 (I do not have other devices at my disposal). I have created a quick GIF showcasing the issue. I can also tell that normal input fields (both common HTML as well as Material inputs) work perfectly fine and always open up the virtual keyboard.

![android-autocomplete](https://user-images.githubusercontent.com/3786768/42219257-9b32d64a-7ecb-11e8-8672-ed5cff01d9f6.gif)

**Comment by Ruudi-V** [comment]
I currently have an analogous problem. The bug was introduced, when i added md-min-length="0" to the autocomplete. 
![autocomplete-bug](https://user-images.githubusercontent.com/41464140/42990474-695ae73c-8c0b-11e8-8db7-d0732289e860.gif)

Related code: 
            <md-autocomplete
                md-selected-item="vm.interfaces.selected"
                md-search-text="vm.searchText"
                md-items="item in vm.searchInterfaces()"
                md-item-text="item.name + ' - ' + item.alias"
                id="interface-select"
                md-search-text-change="vm.searchInterfaces()"
                md-min-length="0"
                >

                <md-item-template>
                    <span md-highlight-text="vm.searchText" md-highlight-flags="i">
                         {{item.name}} - {{item.alias}}
                    </span>
                    <md-not-found>
                        Ei leidnud vastet
                    </md-not-found>
                </md-item-template>
            </md-autocomplete>`

**Comment by Ruudi-V** [comment]
What more info could I provide to help you track down the bug?

**Comment by Splaktar** [comment]
@Ruudi-V your CodePen doesn't run for me. I tried to [fix it a bit](https://codepen.io/Splaktar/pen/KBZEdX?editors=1010#0), but it seems to have some external dependencies.

**Comment by Splaktar** [comment]
I tested this again on a Motorola Nexus 6 with Android 7.0.0. I used SwiftKey keyboard and saw no problems or delays at all. Then I tested with GBoard (the Google Keyboard) and there were a few cases where I could get it to take 2-6 seconds before the keyboard would appear. However, this is an issue with Android and GBoard and not AngularJS Material.

I also tested with the autocomplete panel overlapping the virtual keyboard and I did not see the issue above where the keyboard flickers and then dismisses.

I used [this updated CodePen](https://codepen.io/Splaktar/pen/PBELKb?editors=1010#0) for testing and added the `md-min-length="0"` mentioned above, but it didn't reproduce the issue for me.

**Comment by Ruudi-V** [comment]
Tried your updated [CodePen](https://codepen.io/Splaktar/pen/PBELKb?editors=1010#0) and was able to recreate the issue. 

When logging with [System Info for Android](https://play.google.com/store/apps/details?id=com.electricsheep.asi&hl=en) and spamming the autocomplete I saw the following error:
![logs](https://user-images.githubusercontent.com/41464140/43449869-2cd0dd46-94ba-11e8-8991-da1c13246e64.jpg)



![ezgif com-video-to-gif](https://user-images.githubusercontent.com/41464140/43448925-d1b1c6ac-94b7-11e8-99c7-bbfac5495093.gif)

**Comment by Splaktar** [comment]
@Ruudi-V those logs are very helpful, thank you. Please submit a bug against Chrome for Android [here](https://bugs.chromium.org/p/chromium/issues/list). It looks like Chrome for Android is not handling this error properly when the device is low on memory. It's possibly also related to having hardware acceleration disabled.

**Comment by Ruudi-V** [comment]
Created a chrome bug issue here:
https://bugs.chromium.org/p/chromium/issues/detail?id=869764

**Comment by Splaktar** [comment]
@Ruudi-V can you please copy paste the text from those logs (not just a screenshot) into the Chromium bug please? That is the key indicator that this is a Chrome for Android bug. Without that, they may just dismiss the issue.

**Comment by Splaktar** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=869764 was closed by the Chromium team as part of a bulk issue closing earlier this week. @Ruudi-V are you able to re-open the issue?

**Comment by andresgardiol** [comment]
Hi, I have the same problem on my Xiaomi A2 lite with Android 9. When I enter the Angularjs Material page it does not show the virtual keyboard when I click on the first and second input of the autocomplete demo.
I'm using it in my application and it does not work properly either.

**Comment by Splaktar** [comment]
As a response to my previous comment, I was able to get the Chromium team to reopen the issue.

@andresgardiol can you please add your phone model, android version, keyboard, and Chrome version to a comment in the [Chromium bug](https://bugs.chromium.org/p/chromium/issues/detail?id=869764)? It would also be great if you could share a screen recording so that the Chromium team can continue debugging this.

If anyone else is still experiencing this with a recent version of Chrome, they are asking for
> if anyone could repo on recent Chrome releases and share the new info including Android/keyboard/screen record with us.

**Comment by andresgardiol** [comment]
Ok, I already commented on the problem in the link that you send me. Thank you

**Comment by andresgardiol** [comment]
Was there any news about this issue?

**Comment by Splaktar** [comment]
Nope, it's blocked by the Chromium team. They did identify that it was a regression in Chrome 62.
